workflow-pacta-cop.Rmd

```diff
@@ -0,0 +1,516 @@
+
+4 Peer files are required:
+
+- [project_code]_peers_bonds_results_portfolio.rda
+
+- [project_code]_peers_equity_results_portfolio.rda
+
+- [project_code]_peers_bonds_results_portfolio_ind.rda
+
+- [project_code]_peers_equity_results_portfolio_ind.rda
+
+These should be saved in the folder pacta_data/[timestamp] with the above format. 
+
+These represent two types of aggregations; the aggregation at Investor level for each participant, used to create the rankings and comparative bar charts; and the aggregation at peer group level (bank, pensionfund etc)
+
+The standard peer groups are as follow:
+
+- Pensionfund
+
+- Insurance
+
+- Bank
+
+- Asset manager. 
+
+This is entered by the user on the platform. If you require different peer groups selection options get in contact with Wim, or an alternative combination of these for your results this can be programmed into create_interactive_report. 
+
+### Creating the peer group files
+
+First the individual results are created, then the peer group results.
+
+This all works by using the cleaned portfolio files provided. The format should be one large file, that includes all portfolios appended into one. Include the peer groups in this file for ease. 
+
+Columns:
+Peer.Group; Investor.Name; Portfolio.Name; ISIN; MarketValue; Currency
+
+```{r echo = FALSE}
+
+library("dplyr")
+library("readr")
+
+# open the raw portfolio file here
+portfolio_file <- read_csv("C:/Users/clare/Desktop/sample_cop_file.csv")
+
+req_cols <- c("Peer.Group", "Investor.Name", "Portfolio.Name", "ISIN", "MarketValue", "Currency")
+
+if(length(setdiff(colnames(portfolio_file), req_cols)) > 0){print("Missing or excessive columns in portfolio file")}
+
+portfolio_peers <- portfolio_file %>% 
+  mutate(Investor.Name = Peer.Group,
+         Portfolio.Name = Peer.Group) %>% 
+  select(-Peer.Group)
+
+portfolio_ind <- portfolio_file %>% 
+  select(-Portfolio.Name) %>% 
+  rename(Portfolio.Name = Investor.Name,
+         Investor.Name = Peer.Group) 
+
+
+# Feel free to manually do this step if simpler for you. 
+
+```
+
+The first peer "project" is ready to be run through the analysis with the aggregation at the individual level. 
+
+```{r echo = FALSE}
+
+devtools::load_all()
+use_r_packages()
+## Project Initialisation
+working_location <- paste0(getwd(), "/")
+source("0_portfolio_test.R")
+source("0_graphing_functions.R")
+source("0_reporting_functions.R")
+source("0_portfolio_input_check_functions.R")
+source("0_global_functions.R")
+source("0_sda_approach.R")
+
+
+project_name <- paste0(project_code, "_peers_ind")
+twodii_internal <- FALSE
+
+# TRUE or FALSE: TRUE means that the code is running on a 2dii laptop with dropbox connection
+#####################################################################
+### ONLY FOR EXTERNAL PROJECTS (twodii_internal <- FALSE):
+# Variables must exist for internal projects
+project_location_ext <- "C:/Users/clare/Desktop/ExternalTest"
+data_location_ext <- "C:/Users/clare/Desktop/TestData"
```

Comment:
these will also only work for @Clare2D
I suggest that if there are external files/paths that are needed for this path, that we set them all together, at the very top of this script so it's super clear to a new user "I have to set these first, then I can proceed"

Reply:
I had some troubles with this actually. Because I basically set two projects to run, there was an error that was only resolved by resetting these file paths. I will improve this anyway.